File: day-1/part-2.js
const { read } = require('../utils')

;(async () => {
  const data = (await read('./data.txt')).map(number => Number(number))
  
  for (let i = 0; i < data.length - 3; i++) {
    for (let j = 1; j < data.length - 2; j++) {
      const requiredNum = data.find((number, index) => data[i] + data[j] + number === 2020 && index !== i && index !== j)
      if (requiredNum) {
        console.log('match found:', data[i] * data[j] * requiredNum)
        return
      }
    }
  }
})()

File: day-1/part-2.py
import sys

with open('data.txt') as f:
  data = [int(number) for number in f.read().splitlines()]

  for i in range(len(data) - 3):
    for j in range(1, len(data) - 2):
      requiredNum = 2020 - data[i] - data[j]
      for index, num in enumerate(data):
        if index != i and index != j and num == requiredNum:
          print(data[i] * data[j] * requiredNum)
          sys.exit()

